client/quota.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+type postQuotaData struct {
+	GroupName string   `json:"group-name"`
+	Parent    string   `json:"parent,omitempty"`
+	Snaps     []string `json:"snaps,omitempty"`
+	MaxMemory int64    `json:"max-memory"`
```

Comment:
should this be `uint64` (negative memory doesn't have a meaning) ?

Reply:
Good point, done, although we have inconsistencies elsewhere in existing helpers, so unfortunately there is now int64/uint64 casting (cleaning the helpers up is probably a material for a followup since it would affect unrelated code).

Reply:
sounds good, I think cleanups to the helpers would be good but yeah are unnecessary at this point